### PR TITLE
Enable the registry to be built with debug symbols

### PR DESCRIPTION
**Pull request by RichardScothern**
Enable the registry to be built with debug symbols with an environment variable 'REGISTRY_DEBUG'.

$ REGISTRY_DEBUG=true make binaries

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,6 +4,9 @@ PREFIX?=$(shell pwd)
 # Used to populate version variable in main package.
 VERSION=$(shell git describe --match 'v[0-9]*' --dirty='.m' --always)
 GO_LDFLAGS=-ldflags "-X `go list ./version`.Version $(VERSION)"
+ifeq (${REGISTRY_DEBUG},true)
+	GO_GCFLAGS=-gcflags "-N -l"
```
Comment (dmp42) [inline Makefile]: What does `-l` do?

Comment (RichardScothern) [inline Makefile]: These flags are passed to 6c and prevent inlining functions and storing
variables in registers, making go programs more debuggable in gdb.

**Comment by dmp42** [comment]
LGTM

File: Makefile [hunk 2]
```diff
@@ -18,7 +21,7 @@ version/version.go:
 
 ${PREFIX}/bin/registry: version/version.go $(shell find . -type f -name '*.go')
 	@echo "+ $@"
-	@go build -tags "${DOCKER_BUILDTAGS}" -o $@ ${GO_LDFLAGS} ./cmd/registry
+	@go build -tags "${DOCKER_BUILDTAGS}" -o $@ ${GO_LDFLAGS}  ${GO_GCFLAGS} ./cmd/registry
```
Comment (stevvooe) [inline Makefile]: This should be added to the other invocations of build.

File: Makefile (hunk 1 above)
Comment (stevvooe) [inline Makefile]: I cannot find the documentation for `-l`...

Comment (stevvooe) [inline Makefile]: Let's call this `DISABLE_GC_OPTIMIZATION` or something like that. When this is enabled, something like `-noopts` should be added to the version string.

File: Makefile (hunk 2 above)
Comment (RichardScothern) [inline Makefile]: Done

**Comment by RichardScothern** [comment]
@dmp42
 `go tool 6g` for the flag descriptions

**Comment by stevvooe** [comment]
LGTM

**Pull request merged**